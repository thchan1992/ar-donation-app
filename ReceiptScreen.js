import {View, Text, Button} from 'react-native';
import React from 'react';
const ReceiptScreen = ({navigation, route}) => {
  return (
    <View>
      <Text>Thanks for your donation</Text>
      <Button
        title="Completed"
        onPress={() => {
          navigation.replace('AR');
        }}
      />
    </View>
  );
};

export default ReceiptScreen;
