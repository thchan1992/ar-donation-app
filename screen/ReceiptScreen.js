import React from 'react';
import {Text, StyleSheet, View} from 'react-native';
import PrimaryButton from '../component/PrimaryButton';
import {instrucText} from '../util/styling';
import {completedMsg} from '../constants/msg';

const ReceiptScreen = ({navigation, route}) => {
  const handleCompleted = () => {
    navigation.replace('AR');
  };
  return (
    <View style={styles.container}>
      <Text style={instrucText.instrucText}>
        {completedMsg(route.params.charity.name, route.params.donateAmount)}
      </Text>
      <PrimaryButton text={'Done'} onPress={handleCompleted} />
    </View>
  );
};

export default ReceiptScreen;

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
  },
});
