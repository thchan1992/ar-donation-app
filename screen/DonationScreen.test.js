import React from 'react';
import {create} from 'react-test-renderer';
import ReceiptScreen from './ReceiptScreen';

describe('ReceiptScreen', () => {
  const navigation = {
    replace: jest.fn(),
  };

  it('navigates to AR screen on press of Done button', () => {
    const route = {
      params: {
        charity: {name: 'Test Charity'},
        donateAmount: 10,
      },
    };
    const tree = create(
      <ReceiptScreen navigation={navigation} route={route} />,
    ).root;
    const instance = tree.findByProps({testID: 'doneButton'});
    instance.props.onPress();
    expect(navigation.replace).toHaveBeenCalledWith('AR');
  });
});
