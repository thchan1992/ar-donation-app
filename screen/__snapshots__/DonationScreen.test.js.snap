// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DonationScreen renders correctly 1`] = `
<RCTScrollView>
  <View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "height": 850,
          "paddingTop": 10,
        }
      }
    >
      <Image
        source="image.png"
        style={
          Object {
            "borderColor": "#FF304F",
            "borderRadius": 8,
            "borderWidth": 3,
            "height": 300,
            "padding": 30,
            "paddingTop": 40,
            "width": 300,
          }
        }
      />
      <Text
        style={
          Object {
            "alignSelf": "center",
            "color": "white",
            "fontSize": 25,
            "fontWeight": "bold",
            "padding": 5,
          }
        }
      >
        Please insert the amount you would like to donate to Test Charity. Please make sure the amount should not be larger than £1000
      </Text>
      <View
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "center",
          }
        }
      >
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 50,
                "paddingTop": 20,
              }
            }
          >
            £
          </Text>
          <TextInput
            keyboardType="numeric"
            maxLength={6}
            onChangeText={[Function]}
            placeholder="Amount"
            placeholderTextColor="grey"
            style={
              Array [
                Object {
                  "borderColor": "#FF304F",
                  "borderRadius": 8,
                  "borderWidth": 1,
                  "color": "white",
                  "fontSize": 50,
                  "margin": 12,
                  "maxWidth": 200,
                  "padding": 10,
                },
                Object {
                  "fontSize": 40,
                  "maxWidth": 160,
                  "minWidth": 160,
                },
              ]
            }
            textAlign="center"
            value={null}
          />
        </View>
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          nativeID="animatedComponent"
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "#28C7FA",
              "borderColor": "#FF304F",
              "borderRadius": 8,
              "borderWidth": 1.5,
              "height": 70,
              "margin": 10,
              "opacity": 1,
              "width": 150,
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "alignSelf": "center",
                  "fontSize": 30,
                  "fontWeight": "bold",
                  "padding": 16,
                },
                Object {
                  "alignSelf": "center",
                  "fontSize": 30,
                  "fontWeight": "bold",
                  "padding": 16,
                },
              ]
            }
          >
            Donate
          </Text>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
